package.json

```diff
@@ -9,9 +9,11 @@
       "default": "./lib/index.js"
     },
     "./native": {
+      "types": "./lib/native/index.d.ts",
       "default": "./lib/native/index.js"
```

Comment:
Do you encounter an issue that these types aren't being picked up?

We have the typings mapped explicitly in respective `package.json` files:

https://github.com/mswjs/msw/blob/6931c0ecadca5d4b156c1299bb402f074a3d1a9a/node/package.json#L4

https://github.com/mswjs/msw/blob/6931c0ecadca5d4b156c1299bb402f074a3d1a9a/native/package.json#L4

Reply:
It doesn't look like typescript uses the nested `package.json` file. I quickly tested the solution above with a `.mjs` file and typescript correctly resolves everything.

Reply:
That's interesting. Perhaps #1383 affected how TS sees things. This would be great to turn into an integration test so we could catch broken typings. 

We do have some typings tests but, apparently, they didn't catch this issue?

https://github.com/mswjs/msw/blob/6931c0ecadca5d4b156c1299bb402f074a3d1a9a/test/typings/index.test-d.ts#L2-L3

I'd expect these imports to yield `any` if the typings map is broken, failing the typings tests.

Reply:
Well, those tests can pass or fail depending on the settings inside `tsconfig.json`. 
I can try a few things over the weekend to see in what scenarios it is woorking and where not.

Reply:
I have added some tests that test the moduleResultion of node 16.
I was wrong before. This PR does not fix #1267, but it fixes using `msw/node` in a `.mts` file